core/src/al/miniaudio_device.cpp

```diff
@@ -0,0 +1,238 @@
+#define MINIAUDIO_IMPLEMENTATION
+#include <cubos/core/al/miniaudio_device.hpp>
+#include <cubos/core/log.hpp>
+#include <cubos/core/reflection/external/string.hpp>
+
+using namespace cubos::core::al;
+
+MiniaudioBuffer::MiniaudioBuffer(const void* data, size_t dataSize)
+{
+    if (ma_decoder_init_memory(data, dataSize, nullptr, &mDecoder) != MA_SUCCESS)
+    {
+        CUBOS_CRITICAL("Failed to initialize Decoder from data");
+        abort();
+    }
+}
+
+size_t MiniaudioBuffer::getLength()
+{
+    ma_uint64 lengthInPCMFrames;
+    ma_result result = ma_decoder_get_length_in_pcm_frames(&mDecoder, &lengthInPCMFrames);
+
+    if (result != MA_SUCCESS)
+    {
+        CUBOS_ERROR("Failed to get the length of audio in PCM frames.");
+        return 0;
+    }
+
+    // Calculate the length in seconds: Length in PCM frames divided by the sample rate.
+    return static_cast<size_t>(lengthInPCMFrames) / mDecoder.outputSampleRate;
+}
+
+MiniaudioBuffer::~MiniaudioBuffer()
+{
+    ma_decoder_uninit(&mDecoder);
+};
+
+MiniaudioSource::MiniaudioSource()
+{
+    if (ma_engine_init(nullptr, &mEngine) != MA_SUCCESS)
+    {
+        CUBOS_CRITICAL("Failed to initialize miniaudio engine.");
+        abort();
+    }
```

Comment:
Isn't this function used to initialize the library? If so, then you shouldn't be creating it for every audio source created, right?
You could just create it once in the audio device itself, and pass a reference to it to this class on creation, so that it can use it to do the operations it requires.